wallet/src/de/schildbach/wallet/service/BlockchainServiceImpl.java

```diff
@@ -239,13 +244,17 @@ public void onThrottledWalletChanged() {
         public void onCoinsReceived(final Wallet wallet, final Transaction tx, final Coin prevBalance,
                 final Coin newBalance) {
 
-            final int bestChainHeight = blockChain.getBestChainHeight();
-            final boolean replaying = bestChainHeight < config.getBestChainHeightEver();
+            //final int bestChainHeight = blockChain.getBestChainHeight();
+            final boolean replaying = blockchainState.getReplaying();
```

Comment:
This fix prevents old transactions from being assigned current exchange rates.  

For some reason `bestChainHeight < config.getBestChainHeightEver();` was false when restoring a wallet.  Instead, we will check the most recently saved `blockchainState` to see if it is replaying.

I wonder though, if the app crashes, will replaying be true or false when the app resumes the blockchain sync?  There may still be some holes in this fix.  Need to investigate further why the original code wasn't working correctly.